### DISCUSSION TITLE
Test extension is getting instantiated multiple times when parallel mode is enabled.

### DISCUSSION THREAD

**Discussion by donahchoo** [junit-team/junit-framework#3216] [Q&A]
I'm trying to create a test extension that gets instantiated only once so that my application will only get started once in parallel mode.  When I annotate with `@ExtendWith(MyTestExtension.class)` I can see that MyTestExtension is being instantiated multiple times, at least once per thread.  So far I have not been able to figure out what is causing this behavior or how to work around it.  I've tried using synchronized and volatile but since MyTestExtension is being instantiated multiple time not of that seems to work.

**Comment by jbduncan** [comment]
@donahchoo You can give StackOverflow answer https://stackoverflow.com/a/51556718 a go, or you can try the [resources extension](https://junit-pioneer.org/docs/resources/) I authored recently for junit-pioneer.

I hope this helps.
